src/circuit/lookup.rs

```diff
@@ -20,13 +31,12 @@ fn synth<'a, E: Engine, I>(
     assert_eq!(assignment.len(), 1 << window_size);
 
     for (i, constant) in constants.into_iter().enumerate() {
-        let mut cur = assignment[i];
-        cur.negate();
-        cur.add_assign(constant);
-        assignment[i] = cur;
+        assignment[i].add_assign(constant);
+        let cur = assignment[i];
```

Comment:
We could remove this line, and inline `assignment[i]` in the one place it is used.

Reply:
I really wanted to :) But there's some mutability issue - `assignment` is borrowed as mutable in the inner loop, so you're not allowed to use `assignment[i]` as input to `sub_assign` cause it needs to be immutable.

Reply:
I like it as it is. It's useful to know that ``cur`` is immutable, i.e. that none of the ``eval.sub_assign``s modify it, which would be true but non-obvious if ``assignment[i]`` were inlined. To make it even clearer we could rename ``cur`` to ``assignment_i``.